Sources/MongoSwift/Operations/ListDatabasesOperation.swift

```diff
@@ -17,28 +17,43 @@ public struct DatabaseSpecification: Codable {
 }
 
 /// Internal intermediate result of a ListDatabases command.
-internal enum ListDatabasesResults {
+public enum ListDatabasesResults {
     /// Includes the names and sizes.
     case specs([DatabaseSpecification])
 
     /// Only includes the names.
     case names([String])
 }
 
+/// Options for "listDatabases" operations
+public struct ListDatabasesOptions: Encodable {
+    /// Specifies whether to only return databases for which the user has privileges,
+    /// if false MongoDB will return an error for unauthorized users.
+    public let authorizedDatabases: Bool
+
+    public init(_ authorizedDatabases: Bool = true) {
+        self.authorizedDatabases = Bool(authorizedDatabases)
+    }
+}
+
 /// An operation corresponding to a "listDatabases" command on a collection.
 internal struct ListDatabasesOperation: Operation {
     private let client: MongoClient
     private let filter: Document?
-    private let nameOnly: Bool?
+    private let options: ListDatabasesOptions?
+    /// Specifies whether to only return the list of database names
+    public let nameOnly: Bool?
 
     internal init(
         client: MongoClient,
         filter: Document?,
-        nameOnly: Bool?
+        nameOnly: Bool?,
+        options: ListDatabasesOptions? = nil
```

Comment:
nit: this probably doesn't need a default value, since we already handle the default value of nil at the API level